app/src/main/java/ch/epfl/skysync/dataModels/calendarModels/CalendarModel.kt

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.skysync.dataModels.calendarModels
+
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+abstract class CalendarModel<T: CalendarViewable>{
+    protected val cells: MutableList<T> = mutableListOf()
+    var size: Int = cells.size
+        private set
+
+    abstract fun initForRange(from: LocalDate,
+                              to: LocalDate,)
+
+    /**
+     * @param from inclusive
+     * @param to inclusive
+     */
+    protected fun initForRangeSuper(from: LocalDate,
+                                    to: LocalDate,
+                                    constructor: (LocalDate, TimeSlot) -> T) {
+        val numberOfDays = from.until(to, ChronoUnit.DAYS)
+        var currentDate = from
+        for (i in 0.. numberOfDays) {
+            for (timeSlot in TimeSlot.entries) {
+                cells.add(constructor(currentDate, timeSlot))
+                update_size()
+            }
+            currentDate = currentDate.plusDays(1)
+        }
+
+    }
+    private fun update_size(){
```

Comment:
Instead of creating an attribute `size` and update it each time `cells` is updated, why not have a method `getSize()` (or just `size()`):
```kotlin
fun size(): Int {
    return cells.size
}
```

Reply:
très bonne commentaire, merci